app/core/service/course/course.service.js

```diff
@@ -3,53 +3,36 @@
 angular
     .module('course')
     .factory('courseService', [ '$http', function($http) {
-
+        var courses = [];
         function loadCourses() {
             return $http
                 .get('data/courses.json')
                 .then(function(coursesData) {
-                    angular.forEach(coursesData.data, function(course) {
-                        addDisplayDateAndDuration(course);
-                    });
-
-                    return coursesData.data;
+                    courses = coursesData.data;
                 });
         }
-        function addCourse(data, array) {
-            array.unshift(data);
-        }
-
-        function editCourse(data, array) {
-            addDisplayDateAndDuration(data);
-            var index = array.indexOf(data.selectedCourse);
-            if (index !== -1) array[index] = data;
-        }
 
-        function deleteCourse(course, array) {
-            var index = array.indexOf(course);
-            if (index !== -1) array.splice(index, 1);
+        function addCourse(data) {
+            courses.unshift(data);
         }
 
-        function addDisplayDateAndDuration(course) {
-            var uploadDate = new Date(course.uploadDate);
-            course.displayDate = dateToDisplayDate(uploadDate);
-            course.displayDuration = toDisplayDuration(course.duration);
-            course.uploadDate = uploadDate;
-            course.uploadTime = uploadDate.getTime();
+        function getCourses() {
+            return courses;
         }
 
-        function dateToDisplayDate(date) {
-            return ((date.getMonth() + 1) < 10 ? '0'+ (date.getMonth() + 1) : (date.getMonth() + 1)) + '.' +
-                (date.getDate() < 10 ? '0'+ date.getDate() : date.getDate()) + '.' +
-                date.getFullYear();
+        function editCourse(data) {
```

Comment:
You have strange duplicated structure in you data argument:
![image](https://user-images.githubusercontent.com/25908620/68126504-05577000-ff25-11e9-9a28-b65ae4ea0c18.png)
As a result, course item in array, that is overridden by your  editCourse function, contains selectedCourse field.

Reply:
i used this field for indexOf() place to replace my old object before edit on new object after sending edit form

Reply:
fixed with using lodash